src/python/py/models/builder.py

```diff
@@ -49,19 +50,21 @@ def __init__(self, config, io_dtype, onnx_dtype, ep, cache_dir, extra_options):
         self.cache_dir = cache_dir
         self.filename = extra_options["filename"] if "filename" in extra_options else "model.onnx"
         self.extra_options = extra_options
-        
+
         self.inputs = []
         self.outputs = []
         self.initializers = []
         self.value_infos = []
         self.nodes = []
 
         # Map input names to their types and shapes
-        self.input_names = ["input_ids", "attention_mask", "position_ids"]
+        self.input_names = ["input_ids", "attention_mask", "position_ids"] if not self.enable_cuda_graph else ["input_ids", "position_ids", "seqlens_k", "total_seq_len"]
```

Comment:
Can this conditional logic be moved after the below as another branch case instead?
https://github.com/microsoft/onnxruntime-genai/blob/2fb0c6dacc46169dcbbdeb560492389efa639e6d/src/python/py/models/builder.py#L82-L84